File: Striver's SDE Sheet/sortColors/code.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int>sortColors(vector<int>&nums){
    int n=nums.size();
    int i=0,j=0,k=n-1;

    while(j<=k){
        if(nums[j]==0){
            swap(nums[i],nums[j]);
            i++;
            j++;
        }
        else if(nums[j]==1){
            j++;
        }
        else{
            swap(nums[j],nums[k]);
            k--;
        }
    }
    return nums;
}

int main(){
    vector<int>nums={2,0,2,1,1,0};
    vector<int>ans=sortColors(nums);

    for(int num:ans){
        cout<<num<<" ";
    }
    cout<<endl;
}

File: Striver's SDE Sheet/sortColors/code.py
def sort_colors(nums):
    n=len(nums)
    i=0
    j=0
    k=n-1

    while(j<=k):
        if(nums[j]==0):
            nums[i],nums[j]=nums[j],nums[i]
            i+=1
            j+=1
        elif(nums[j]==1):
            j+=1
        else:
            nums[j],nums[k]=nums[k],nums[j]
            k-=1
    
    return nums

nums=[2,0,2,1,1,0]
ans=sort_colors(nums)
print(" ".join(map(str,ans)))